iothub/service/design.md

```diff
@@ -0,0 +1,119 @@
+# RBAC API Design for IoT hub
+
+## Service Client
+
+```csharp
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
```

Comment:
I know that this is from existing code comment, but do we want to specify which operations use http on the service client? This is only because in the transportType above we ask about amqp, but then in proxy settings we talk about both amqp and http.
I know that this is such a bad design, and looking at it for the first time confused me; so I wanted to know what you think about it.

Reply:
I think it might be too much to specify that info here in comments. I'm thinking how about we create a .md that talks about it in detail and just link it here?

Reply:
I can add that as a task on my card.

Reply:
Works for me, it might be a lot of info for doc comments, true. 
small nit: update AMQP to AMQP_WS